Include info for when a trace is truncated (#29245)
dotnet/docs: docs/core/diagnostics/dotnet-trace.md

  Shows the input and output streams of a launched child process in the current console.

> [!NOTE]

> - Stopping the trace may take a long time (up to minutes) for large applications. The runtime needs to send over the type cache for all managed code that was captured in the trace.

> - On Linux and macOS, this command expects the target application and `dotnet-trace` to share the same `TMPDIR` environment variable. Otherwise, the command will time out.

> > - To collect a trace using `dotnet-trace`, it needs to be run as the same user as the user running target process or as root. Otherwise, the tool will fail to establish a connection with the target process.

Reviewer: ```suggestion
> - To collect a trace using `dotnet-trace`, it needs to be run as the same user as the user running the target process or as root. Otherwise, the tool will fail to establish a connection with the target process.
```

> > - If you see an error message similar to: `[ERROR] System.ComponentModel.Win32Exception (299): A 32 bit processes cannot access modules of a 64 bit process.`, you are trying to use `dotnet-trace` that has mismatched bitness against the target process. Make sure to download the correct bitness of the tool in the [install](#install) link.

Reviewer: ```suggestion
> - If you see an error message similar to: `[ERROR] System.ComponentModel.Win32Exception (299): A 32 bit processes cannot access modules of a 64 bit process.`, you are trying to use a version of `dotnet-trace` that has mismatched bitness against the target process. Make sure to download the correct bitness of the tool in the [install](#install) link.
```

> > - If you experience an unhandled exception while running `dotnet-trace collect` this results in a broken trace. If root-causing the exception is your priority, navigate to [Collect dumps on crash](/core/diagnostics/dumps#collect-dumps-on-crash). As a result of the crash in the program, the trace is truncated when the runtime rips apart to prevent breaking other parts of the program. Even though the trace is broken, it can still be opened to see what happened leading up to the failure, but will be missing Rundown information (this happens at the end of a trace) so stacks migt be unresolved (depending on what providers were turned on). Opening the trace is done by executing PerfView with the /ContinueOnError flag at the command line. The logs will also contain the location the exception was fired.

Reviewer: ```suggestion
> - If you experience an unhandled exception while running `dotnet-trace collect`, this results in a broken trace. If finding the root cause of the exception is your priority, navigate to [Collect dumps on crash](/core/diagnostics/dumps#collect-dumps-on-crash). As a result of the crash in the program, the trace is truncated when the runtime rips apart to prevent breaking other parts of the program. Even though the trace is broken, you can still open it to see what happened leading up to the failure. However, will be missing Rundown information (this happens at the end of a trace) so stacks might be unresolved (depending on what providers were turned on). Open the trace by executing PerfView with the `/ContinueOnError` flag at the command line. The logs will also contain the location the exception was fired.
```

> [!NOTE]

> > If you see an error message similar to the following one: `[ERROR] System.ComponentModel.Win32Exception (299): A 32 bit processes cannot access modules of a 64 bit process.`, you are trying to use `dotnet-trace` that has mismatched bitness against the target process. Make sure to download the correct bitness of the tool in the [install](#install) link.

Reviewer: ```suggestion
> - If you see an error message similar to `[ERROR] System.ComponentModel.Win32Exception (299): A 32 bit processes cannot access modules of a 64 bit process.`, you are trying to use a verison of `dotnet-trace` that has mismatched bitness against the target process. Make sure to download the correct bitness of the tool in the [install](#install) link.
> - If you experience an unhandled exception while running `dotnet-trace collect`, this results in a broken trace. If finding the root cause of the exception is your priority, see [Collect dumps on crash](dumps.md#collect-dumps-on-crash). As a result of the crash in the program, the trace is truncated when the runtime rips apart to prevent breaking other parts of the program. Even though the trace is broken, it can still be opened to see what happened leading up to the failure. But it will be missing Rundown information (this happens at the end of a trace), so stacks might be unresolved (depending on what providers were turned on). To open the trace, execute PerfView with the `/ContinueOnError` flag at the command line. The logs will also contain the location where the exception was fired.
```

> > [!NOTE]

Reviewer: ```suggestion
```

> > If you experience an unhandled exception while running `dotnet-trace collect` this results in a broken trace. If root-causing the exception is your priority, navigate to [Collect dumps on crash](https://docs.microsoft.com/en-us/dotnet/core/diagnostics/dumps#collect-dumps-on-crash). As a result of the crash in the program, the trace is truncated when the runtime rips apart to prevent breaking other parts of the program. Even though the trace is broken, it can still be opened to see what happened leading up to the failure, but will be missing Rundown information (this happens at the end of a trace) so stacks migt be unresolved (depending on what providers were turned on). Opening the trace is done by executing PerfView with the /ContinueOnError flag at the command line. The logs will also contain the location the exception was fired.

Reviewer: ```suggestion
```
